app/controllers/admin_controller.rb

```diff
@@ -123,6 +123,23 @@ def mark_spam
     end
   end
 
+  def mark_comment_spam
+    @comment = Comment.find params[:id]
+    if current_user && (current_user.role == 'moderator' || current_user.role == 'admin')
+      if @comment.status == 0
+        @comment.comment_spam
+        flash[:notice] = "Comment has been marked as spam."
+        redirect_to '/dashboard'
```

Comment:
IMHO, since the user is redirected in any case, the redirect statement can be placed at the end of the action (just after line 140).

Reply:
Hi @seafr, yes you are absolutely right, actually, it's just a basic function to show @jywarren to get approval that I am going correctly with the issue and to ask if the status code used is fine or not.But still thanks for highlighting, I will definitely correct it in next commit.